lib/active_storage/service/mirror_service.rb

```diff
@@ -26,9 +26,9 @@ def primary_service
     end
 
     def perform_across_services(method, *args)
-      # FIXME: Convert to be threaded
-      services.collect do |service|
-        service.public_send method, *args
+      threads = services.collect do |service|
+        Thread.new { service.public_send method, *args }
       end
+      threads.collect { |thread| thread.join }
```

Comment:
(Deleted my previous comment because I realized it was wrong after I posted it.)